src/main/java/seedu/foodrem/logic/parser/TagCommandParser.java

```diff
@@ -0,0 +1,46 @@
+package seedu.foodrem.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.foodrem.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.foodrem.logic.parser.CliSyntax.PREFIX_ID;
+import static seedu.foodrem.logic.parser.CliSyntax.PREFIX_TAG_NAME;
+
+import java.util.stream.Stream;
+
+import seedu.foodrem.commons.core.index.Index;
+import seedu.foodrem.logic.commands.tagcommands.TagCommand;
+import seedu.foodrem.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new TagCommand object
+ */
+public class TagCommandParser implements Parser<TagCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the TagCommand
+     * and returns a TagCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public TagCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_TAG_NAME, PREFIX_ID);
+        if (!arePrefixesPresent(argMultimap, PREFIX_TAG_NAME, PREFIX_ID)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, TagCommand.MESSAGE_USAGE));
+        }
+        Index index = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_ID).get());
+        String name = ParserUtil.parseTagName(argMultimap.getValue(PREFIX_TAG_NAME).get()).toString();
+
+        return new TagCommand(name, index);
+    }
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
```

Comment:
I believe this has already been defined in AddCommandParser, shall we move this to e.g. ParserUtil?

Reply:
Yes, good catch. I will refactor this into ParserUtil for AddCommandParser and TagCommandParser to use.